src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -179,5 +206,45 @@ public class Utils {
         r.addAction(new ExecutionModelAction(stages))
     }
 
+    /**
+     * Creates and sets the loading for a cache of {@link Describable}s descending from the given descriptor type.
+     *
+     * @param type The {@link Descriptor} class whose extensions we want to find.
+     * @param excludedSymbols Optional list of symbol names to exclude from the cache.
+     * @return A {@link LoadingCache} for looking up types from symbols.
+     */
+    static generateTypeCache(Class<? extends Descriptor> type, List<String> excludedSymbols = []) {
+        return CacheBuilder.newBuilder()
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build(new CacheLoader<Object, Map<String, String>>() {
+            @Override
+            Map<String, String> load(Object key) throws Exception {
+                return populateTypeCache(type, excludedSymbols)
+            }
+        })
+    }
+
+    /**
+     * Actually populates the type cache.
+     *
+     * @param type The {@link Descriptor} class whose extensions we want to find.
+     * @param excludedSymbols Optional list of symbol names to exclude from the cache.
+     * @return A map of symbols or class names to class names.
+     */
+    private static Map<String,String> populateTypeCache(Class<? extends Descriptor> type, List<String> excludedSymbols = []) {
+        Map<String,String> knownTypes = [:]
+
+        ExtensionList.lookup(type).each { t ->
```

Comment:
In some cases, I'm adding two entries to the map for a single `Extension` - i.e., if the `Extension` has a `Symbol`, I'm adding the symbol *and* the class name as keys.